src/main/java/frc/robot/subsystems/conveyor/ConveyorIO.java

```diff
@@ -0,0 +1,23 @@
+package frc.robot.subsystems.conveyor;
+
+import edu.wpi.first.units.Current;
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.units.Voltage;
+import org.littletonrobotics.junction.AutoLog;
+
+public interface ConveyorIO {
+
+    ConveyorInputsAutoLogged inputs = new ConveyorInputsAutoLogged();
+
+    void setPower(double power);
+
+    void updateInputs();
+
+    @AutoLog
+    class ConveyorInputs {
+        public double setpointPower = 0;
```

Comment:
`powerSetpoint`, although I'd refrain from using "setpoint" here, this term should only be used for closed-loop control contexts.

Reply:
Well how else would I denote that this power is the wanted applied setpoint, and not the actual?

Reply:
0d292c96d376bed30857f9329910fda7f91ac569

Reply:
Ope- loop control is by definition "מה שיוצא אני מרוצה". There is no guarantee that the subsystem will maintain the commanded value, so what conclusions can you get from comparing the actual value to the "setpoint"?
I do think that with how easy closed-loop control is in FRC we can use it everywhere, especially in an important subsystem that feeds the shooter.

Reply:
> There is no guarantee that the subsystem will maintain the commanded value.

That is why we log the actual voltage as well.
I guess for the conveyor we can do PID. For literally everything else it's just overkill.

Reply:
> That is why we log the actual voltage as well.

And I think it's a good idea, but the setpoint value has no meaning here. If it was closed-loop controlled you could see e.g. a difference between the setpoint and the actual value and draw some conclusions ("$k_V$ should be higher"), here you won't be able to see that. Maybe you could see if your current limit is too aggressive, but that's about it.
> For literally everything else it's just overkill.

Maybe, but as I wrote somewhere already, using closed-loop control is as easy as switching a control request, running a SysId routine, and writing the values down. The tuning takes you more than 30 minutes and you didn't receive good results? Post that on the SysId Chief Delphi thread and revert to open-loop.

Reply:
But anywhere except the feeding it just adds some time, and doesn't help at all.

Reply:
Adds some time in development or in match time? If the first, then that's why I suggest capping the time, if the latter, then it's negligible (you might even reach the setpoint faster with closed-loop control).
I think there might be a small advantage in controlling the gripper's speed to make sure the Note is held in the gripper at the same place every time. The same can be said about the intake, so the Note is transferred consistently to the gripper. However, these effects in both subsystems may be unnoticeable.

In the worst case, you're getting some software points from the judges.